build/source/netcdf/modelwrite.f90

```diff
@@ -321,16 +322,235 @@ subroutine writeTime(modelTimestep,outputTimestep,meta,dat,err,message)
     ! get variable id in file
     err = nf90_inq_varid(ncid(iFreq),trim(meta(iVar)%varName),ncVarID) 
     if (err.gt.0) message=trim(message)//trim(meta(iVar)%varName)
-    call netcdf_err(err,message); if (err/=0) then; err=20; return; endif
+    call netcdf_err(err,message); if (err/=0) then; err=20; return; end if
 
     ! add to file
     err = nf90_put_var(ncid(iFreq),ncVarID,(/dat(iVar)/),start=(/outputTimestep(iFreq)/),count=(/1/))
     if (err.gt.0) message=trim(message)//trim(meta(iVar)%varName)
-    call netcdf_err(err,message); if (err/=0) then; err=20; return; endif
+    call netcdf_err(err,message); if (err/=0) then; err=20; return; end if
 
-   enddo ! iVar
-  enddo ! iFreq
+   end do ! iVar
+  end do ! iFreq
 
  end subroutine writeTime 
 
+ ! *********************************************************************************************************
+ ! private subroutine printRestartFile: print a re-start file
+ ! *********************************************************************************************************
+ subroutine writeRestart(filename,         & ! intent(in): name of restart file
+                         nGRU,             & ! intent(in): number of GRUs
+                         nHRU,             & ! intent(in): number of HRUs
+                         prog_meta,        & ! intent(in): prognostics metadata 
+                         prog_data,        & ! intent(in): prognostics data
+                         indx_meta,        & ! intent(in): index metadata 
+                         indx_data,        & ! intent(in): index data
+                         err,message)        ! intent(out): error control
+ ! --------------------------------------------------------------------------------------------------------
+ ! --------------------------------------------------------------------------------------------------------
+ ! access the derived types to define the data structures
+! USE data_types,only:hru_d                  ! length of substep
```

Comment:
comment line `! USE data_types,only:hru_d                  ! length of substep` can be deleted if it is not used.